### ISSUE TITLE
Please don't implement a feature suggested in a categorical cross-entropy TODO

### ISSUE THREAD

**Issue by JimBiardCics** [tensorflow/tensorflow#34462]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): v2.0.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
This is actually an anti-request. I'm requesting that the developers not implement suggested future behavior.

There is a TODO comment in [tensorflow/python/ops/nn_ops.py line 3289](https://github.com/tensorflow/tensorflow/blob/b646d41f69f8a1cf2f5c76d934adbf8c067e4444/tensorflow/python/ops/nn_ops.py#L3289) that states an intention to change the categorical cross-entropy loss algorithm so that it will raise an error if the sum of any label vector is not equal to 1. The comment states:
```
  # TODO(pcmurray) Raise an error when the labels do not sum to 1. Note: This
  # could break users who call this with bad labels, but disregard the bad
  # results.
```

I make use of label vectors with all elements set to zero in order to perform spatial masking of data in my 2D convolutional neural network. I have regions in some of my images where there is no label available, and the regions change from image to image. These regions make no contribution to the loss if the corresponding label vectors are set to all zero, effectively removing them from gradient calculation and back-propagation. (The regions in the images have valid input data, but a label cannot be assigned.) My suggestion is that this TODO not be implemented or that the implementation would allow for all-zero label vectors.

**Will this change the current api? How?**
This will not change the current API.

**Who will benefit with this feature?**
People (like me) who use the current way the algorithm is implemented to do per-image spatial masking.

**Comment by amahendrakar** [comment]
@JimBiardCics,
Is this still an issue? 

I see that the implementation has not been changed from [TF v2.0](https://github.com/tensorflow/tensorflow/blob/b646d41f69f8a1cf2f5c76d934adbf8c067e4444/tensorflow/python/ops/nn_ops.py#L3289) to [TF v2.4](https://github.com/tensorflow/tensorflow/blob/86c79e9779a2d2653fddbb0e0391e2ea8121e81c/tensorflow/python/ops/nn_ops.py#L3920). Thanks!
